src/utils/app/file.ts

```diff
@@ -177,3 +184,61 @@ export const getExtensionsListForMimeTypes = (mimeTypes: string[]) => {
     .flat()
     .map((type) => `.${type}`);
 };
+
+export const getFileNameWithoutExtension = (filename: string) =>
+  filename.slice(0, filename.lastIndexOf('.'));
+
+export const getFileNameExtension = (filename: string) =>
+  filename.slice(filename.lastIndexOf('.'));
+
+export const validatePublishingFileRenaming = (
+  files: PublishAttachmentInfo[],
+  newName: string,
+  renamingFile: PublishAttachmentInfo,
+) => {
+  const fileWithSameName = files.find(
+    (file) =>
+      file.name === newName.trim() &&
+      file !== renamingFile &&
+      file.path === renamingFile?.path,
+  );
+
+  if (fileWithSameName) {
+    return 'Not allowed to have folders with same names';
+  }
+
+  if (newName.match(notAllowedSymbolsRegex)) {
+    return `The symbols ${notAllowedSymbols} are not allowed in folder name`;
```

Comment:
I think you meant a file, not a folder.

```suggestion
  if (fileWithSameName) {
    return 'Not allowed to have files with same names';
  }

  if (newName.match(notAllowedSymbolsRegex)) {
    return `The symbols ${notAllowedSymbols} are not allowed in file name`;
```